lib/ansible/module_utils/network/nxos/config/vlans/vlans.py

```diff
@@ -0,0 +1,243 @@
+
+        commands = list()
+        if state == 'overridden':
+            commands.extend(self._state_overridden(want, have))
+        elif state == 'deleted':
+            commands.extend(self._state_deleted(want, have))
+        else:
+            for w in want:
+                if state == 'merged':
+                    commands.extend(self._state_merged(w, have))
+                elif state == 'replaced':
+                    commands.extend(self._state_replaced(w, have))
+        return commands
+
+    def _state_replaced(self, w, have):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        obj_in_have = search_obj_in_list(w['vlan_id'], have, 'vlan_id')
+        diff = dict_diff(w, obj_in_have)
+        merged_commands = self.set_commands(w, have)
+        if 'vlan_id' not in diff:
+            diff['vlan_id'] = w['vlan_id']
+        replaced_commands = self.del_all(diff)
+
+        if merged_commands:
+            cmds = set(replaced_commands).intersection(set(merged_commands))
+            for cmd in cmds:
+                merged_commands.remove(cmd)
+            commands.extend(replaced_commands)
+            commands.extend(merged_commands)
+        return commands
+
+    def _state_overridden(self, want, have):
+        """ The command generator when state is overridden
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        for h in have:
+            obj_in_want = search_obj_in_list(h['vlan_id'], want, 'vlan_id')
+            if h == obj_in_want:
+                continue
+            commands.extend(self.del_all(h))
+        for w in want:
+            commands.extend(self.set_commands(w, have))
+        return commands
+
+    def _state_merged(self, w, have):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        return self.set_commands(w, have)
+
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        commands = []
+        if want:
+            for w in want:
+                obj_in_have = search_obj_in_list(w['vlan_id'], have, 'vlan_id')
+                commands.extend(self.del_all(obj_in_have))
+        else:
+            if not have:
+                return commands
+            for h in have:
+                commands.append('no vlan ' + str(h['vlan_id']))
+        return commands
+
+    def del_all(self, obj):
```

Comment:
Can this method be clubbed with `add_commands` by adding a flag to determine if we need to append a negate?
Like [this](https://github.com/ansible/ansible/pull/59724/files#diff-9b7e2c51c0a9dcacfabf900240548efaR239).